bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/CleanupLedgerManager.java

```diff
@@ -108,64 +115,69 @@ private GenericCallback removeCallback(GenericCallback callback) {
         return callbacks.remove(callback);
     }
 
+    private void recordPromise(CompletableFuture<?> promise) {
```

Comment:
I am not sure we need this or not. The underlying LM should error about its complete futures, so all the complete futures at this class will be errored out correctly. The change here is masking any potentially errors in underlying LM. I would suggest not adding such logic for now. or at least don't do it in a refactor PR.

Reply:
Are you saying the underlying LM should keep track of all futures they return? 
This change is to basically do the same thing as the CleanupLedgerManager did for callbacks. i.e. If close is called, all outstanding futures should be closed, and all future calls also closed. Moving this into the underlying LM would basically mean making an AbstractLM with this recordPromise and pulling that into the other LMs.

Reply:
No, what I mean is underlying implementation will probably already handle that. we don't need to worry about that at this layer.

it is a good feature to have at any case. but let's do it in a separate PR rather than doing it in a large refactor.

Reply:
the underlying implementations don't handle it though. for them to handle it, each would have to have this recording logic. so it's cleaner to have it abstracted somewhere else.

The reason I did it in this patch is so this patch doesn't remove functionality. Currently, the cleanup lm ensures that if the lm is closed, then all outstanding requests are completed. If we don't implement this here, then this functionality is lost.